app/logic/Calendar/Invitation/IncomingInvitation.ts

```diff
@@ -35,25 +40,40 @@ export class IncomingInvitation {
     /* else add participant? */
   }
 
-  /** Handle the other two InvitationMessage cases:
-   *  CancelledEvent: the organiser cancelled an incoming meeting
-   *  ParticpantReply: an invitee replied to your outgoing invitation */
+  /** CancelledEvent: the organiser cancelled an incoming meeting */
+  async updateCancelled() {
+    assert(this.invitationMessage && this.invitationMessage != InvitationMessage.Invitation, "Can't update from an invitation");
+    let event = this.calEvent();
+    // Is this action reversible? If so, need to check timestamp.
+    let organizer = event.participants.find(participant => participant.response == InvitationResponse.Organizer);
+    if (organizer) {
+      organizer.response = InvitationResponse.Decline;
+      await event.save();
+    }
+  }
+
+  /** ParticpantReply: an invitee replied to your outgoing invitation */
+  async updateParticipantReply() {
```

Comment:
```suggestion
  /** ParticpantReply: an invitee replied to your outgoing invitation */
  protected async updateParticipantReply() {
```
(If this is going to be public API then the subclasses need to override it.)